src/Facets/NEARIntentsFacet.sol

```diff
@@ -0,0 +1,331 @@
+    /// @param minAmountOut Minimum output amount on destination (slippage protection)
+    /// @param signature The signature of the NEARIntentsPayload signed by the backend signer using EIP-712 standard
+    struct NEARIntentsData {
+        address depositAddress;
+        bytes32 nonEVMReceiver;
+        bytes32 quoteId;
+        uint256 deadline;
+        uint256 minAmountOut;
+        bytes signature;
+    }
+
+    /// Storage ///
+
+    /// @notice Diamond storage structure (minimal - only replay protection)
+    struct Storage {
+        /// @dev Mapping to prevent duplicate quote usage (quoteId => consumed)
+        mapping(bytes32 => bool) consumedQuoteIds;
+    }
+
+    /// Events ///
+
+    /// @notice Emitted when a bridge operation starts via NEAR Intents
+    /// @notice Required by NEAR off-chain infrastructure to track deposits and initiate intent settlement
+    /// @param transactionId Unique transaction identifier
+    /// @param quoteId NEAR Intents quote identifier
+    /// @param depositAddress Address tokens were sent to
+    /// @param sendingAssetId Token being bridged
+    /// @param amount Amount being bridged
+    /// @param deadline Quote expiration timestamp
+    event NEARIntentsBridgeStarted(
+        bytes32 indexed transactionId,
+        bytes32 indexed quoteId,
+        address indexed depositAddress,
+        address sendingAssetId,
+        uint256 amount,
+        uint256 deadline
+    );
+
+    /// Errors ///
+
+    /// @notice Thrown when trying to use a quote that was already consumed
+    error QuoteAlreadyConsumed();
+
+    /// @notice Thrown when the quote deadline has passed
+    error QuoteExpired();
+
+    /// @notice Thrown when the signature is invalid
+    error InvalidSignature();
+
+    /// Constructor ///
+
+    /// @notice Initializes the NEARIntentsFacet contract
+    /// @param _backendSigner The address of the backend signer
+    constructor(address _backendSigner) {
+        if (_backendSigner == address(0)) {
+            revert InvalidConfig();
+        }
+        BACKEND_SIGNER = _backendSigner;
+    }
+
+    /// Modifiers ///
+
+    /// @dev Validates quote parameters (consolidated from separate helper function)
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _nearData Data specific to NEAR Intents
+    modifier onlyValidQuote(
+        ILiFi.BridgeData memory _bridgeData,
+        NEARIntentsData calldata _nearData
+    ) {
+        Storage storage s = getStorage();
+
+        // Prevent replay attacks
+        if (s.consumedQuoteIds[_nearData.quoteId]) {
+            revert QuoteAlreadyConsumed();
+        }
+
+        // Ensure quote hasn't expired (single check, removed duplicate from signature verification)
+        if (block.timestamp > _nearData.deadline) {
+            revert QuoteExpired();
+        }
+
+        // Ensure deposit address is valid
+        if (_nearData.depositAddress == address(0)) {
```

Comment:
This check is not required since this is already being checked by LibAsset.transferAsset